packages/wonder-stuff-i18n/src/plugins/i18n-plugin.js

```diff
@@ -0,0 +1,410 @@
+ *   directory.
+ */
+const addAsset = (assets, relFilePath, data) => {
+    assets[relFilePath] = {
+        source() {
+            return data;
+        },
+        size() {
+            return data.length;
+        },
+    };
+};
+
+export type TranslatedStrings = {
+    [locale: string]: TranslatedLocaleStrings,
+};
+
+type IncomingOptions = {
+    locales: Array<string>,
+    silent?: boolean,
+    timing?: {
+        start: (string) => void,
+        end: (string) => void,
+        ...
+    },
+    getI18nStrings: (locale: string) => Promise<TranslatedLocaleStrings>,
+    getLocalePath?: (locale: string) => string,
+    shouldLocalizeAsset?: (assetName: string) => boolean,
+    shouldProcessAsset?: (assetName: string) => boolean,
+    shouldLocalizeManifest?: (assetName: string, locale: string) => boolean,
+};
+
+type Options = {
+    locales: Array<string>,
+    silent?: boolean,
+    timing?: {
+        start: (string) => void,
+        end: (string) => void,
+        ...
+    },
+    getI18nStrings: (locale: string) => Promise<TranslatedLocaleStrings>,
+    getLocalePath: (locale: string) => string,
+    shouldLocalizeAsset: (assetName: string) => boolean,
+    shouldProcessAsset: (assetName: string) => boolean,
+    shouldLocalizeManifest: (assetName: string, locale: string) => boolean,
+};
+
+type LocaleHashMap = {[oldHash: string]: string};
+type HashMaps = {[locale: string]: LocaleHashMap};
+
+export type Assets = {
+    [assetName: string]: {|
+        source: () => string,
+        size: () => number,
+    |},
+};
+export default class I18nPlugin {
+    options: Options;
+
+    constructor(options: IncomingOptions) {
+        if (!options.locales) {
+            throw new Error("Must provide locales to localize to");
+        }
+
+        if (!options.getI18nStrings) {
+            throw new Error(
+                "Must provide a function to get the strings to localize",
+            );
+        }
+
+        this.options = {
+            getLocalePath: (locale) => `/${locale}/`,
+            shouldLocalizeAsset: (assetName) =>
+                assetName.startsWith("runtime."),
+            shouldProcessAsset: () => true,
+            shouldLocalizeManifest: (assetName) => assetName.endsWith(".html"),
+            ...options,
+            // Filter out English locale if it exists
+            locales: options.locales.filter((locale) => locale !== "en"),
+        };
+
+        if (this.options.locales.length === 0) {
+            this.log("I18nPlugin: No locales to localize to.");
+        }
+    }
+
+    /* istanbul ignore next */
+    apply(compiler: $FlowFixMe) {
+        compiler.hooks.emit.tapAsync("I18nPlugin", (options, callback) =>
+            this.onEmit(options, callback),
+        );
+    }
+
+    /**
+     * Log out something to the console, if we're not silent.
+     *
+     * @param  {...any} args the args to log out to the console
+     */
+    log(...args: Array<$FlowFixMe>) {
```

Comment:
I think using `Array<any>` is appropriate here.  This is how flow itself types `console.log()`, see https://github.com/facebook/flow/blob/main/lib/core.js#L2453.  Using `$FlowFixMe` may tempt some people to figure out a fix even though there isn't one.

Reply:
Hmm... so we have a rule in this repo that prevents all usage of `any`, which is why I had to change this! Is there a way around this?